pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -391,9 +402,8 @@ func (e *eksOperatorController) updateEKSClusterConfig(cluster *mgmtv3.Cluster,
 	}
 }
 
-// generateAndSetServiceAccount reads the EKSClusterConfig's secret once available and uses its fields to generate a
-// service account token. The token, CA cert, and API endpoint are then copied to the cluster status.
-func (e *eksOperatorController) generateAndSetServiceAccount(cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
+// getCAAndAPIEndpoint reads the EKSClusterConfig's secret once available. The CA cert and API endpoint are then copied to the cluster status.
+func (e *eksOperatorController) getCAAndAPIEndpoint(cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
```

Comment:
Whats the reason for splitting them up, just readability ? Seems like they could all be inline.

Reply:
We can get the `APIEndpoint` and `CACert` before the cluster agent is deployed. Since we can get those things, we need to make sure that they are added to Status (which is what happens at the end of this method). It simplified some error checking and made sure the items were set in Status in case a later error was produced.